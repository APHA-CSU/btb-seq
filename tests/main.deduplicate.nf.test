nextflow_process {

    name "Test Process deduplicate"
    script "main.nf"
    process "deduplicate"

    test("Should generate fastq files from gzipped input files") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = ["tinyreads", file("${projectDir}/tests/data/tinyreads/tinyreads_S1_R1_X.fastq.gz"), file("${projectDir}/tests/data/tinyreads/tinyreads_S1_R2_X.fastq.gz")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
