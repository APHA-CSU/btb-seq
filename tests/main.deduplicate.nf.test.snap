{
    "Should generate fastq files from gzipped input files": {
        "content": [
            {
                "0": [
                    [
                        "tinyreads",
                        "dedup_1.fastq:md5,8a424d8d8ca95e0bd2c55b663a2bebdc",
                        "dedup_2.fastq:md5,d690d4f6f937607963a36d61ac16d28d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-27T17:03:30.654983781"
    }
}